lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -369,12 +369,13 @@ def check_distance(route, visits, train = nil)
         def revenue_for(route, stops)
           revenue = super
 
-          abilities = Array(abilities(route.corporation, :hex_bonus))
-          return revenue if abilities.empty?
+          route.corporation.companies.each do |company|
+            abilities(company, :hex_bonus) do |ability|
```

Comment:
You don't need nested iterators here. Passing a Corporation object to `Game.abilities` will return matching abilities from both the corporation and any companies that it owns. See `Corporation.all_abilities`: https://github.com/tobymao/18xx/blob/38a729965ef394d5cd3ace16e45952672fa64aac/lib/engine/corporation.rb#L252-L262

If you do have the nested iterator, you need to remember that `abilities` can return nil, a single ability, or an array of abilities. The inner iterator will fail unless the company has two or more hex_bonus abilities.

Reply:
done